xchange-poloniex2/src/main/java/info/bitrich/xchangestream/poloniex2/PoloniexStreamingMarketDataService.java

```diff
@@ -90,12 +79,26 @@ public PoloniexStreamingMarketDataService(PoloniexStreamingService service, Map<
                 .share();
 
         return subscribedTrades
-                .map(s -> adaptPoloniexPublicTrade(s.toPoloniexPublicTrade(currencyPair), currencyPair));
+                .map(s -> adaptPoloniexPublicTrade(s, currencyPair));
     }
 
-    private Optional<PoloniexOrderbook> getPoloniexOrderbook(
-            final Optional<PoloniexOrderbook> orderbook, final PoloniexWebSocketEvent s
-    ) {
+    private static Trade adaptPoloniexPublicTrade(
```

Comment:
Could you move this to a static method on a `PoloniexAdapters`class?

Reply:
No, because it's a different library. We can create a different class, but moving it to xchange doesn't make sense.

Reply:
though I'll rename the method. as it doesn't reflect what's going on

Reply:
I didn't mean moving it to XChange. I just meant moving it to a utility class, making it default-visible.  See `BitfinexStreamingAdapters` for an example.

Once there, you can/could write a unit test for it (which I'm not asking you to write, because I'm trying to avoid creating too much extra work for you, but at least it means someone else could in the future and it keeps this class tidy).

Reply:
ok. I'll create a new class for the conversion code